modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ToolbarButtons.ts

```diff
@@ -303,140 +304,183 @@ const fetchChoices = (getApi: (comp: AlloyComponent) => Toolbar.ToolbarSplitButt
+    Traverse.nextSibling(element).bind((el) => system.getByDom(el).toOptional().filter((comp) => Class.has(comp.element, 'tox-split-button__chevron')));
+
+  const applyBoth = (f: (c: AlloyComponent) => void) => {
+    mainOpt.each(f);
+    chevronOpt.each(f);
+  };
 
-  const getApi = (comp: AlloyComponent): Toolbar.ToolbarSplitButtonInstanceApi => ({
-    isEnabled: () => !Disabling.isDisabled(comp),
-    setEnabled: (state: boolean) => Disabling.set(comp, !state),
-    setIconFill: (id, value) => {
-      SelectorFind.descendant(comp.element, `svg path[class="${id}"], rect[class="${id}"]`).each((underlinePath) => {
+  return {
+    isEnabled: () => mainOpt.exists((c) => !Disabling.isDisabled(c)),
+    setEnabled: (state: boolean) => applyBoth((c) => Disabling.set(c, !state)),
+    setText: (text: string) => mainOpt.each((c) => AlloyTriggers.emitWith(c, updateMenuText, { text })),
+    setIcon: (icon: string) => mainOpt.each((c) => AlloyTriggers.emitWith(c, updateMenuIcon, { icon })),
+    setIconFill: (id: string, value: string) => applyBoth((c) => {
+      SelectorFind.descendant(c.element, `svg path[class="${id}"], rect[class="${id}"]`).each((underlinePath) => {
         Attribute.set(underlinePath, 'fill', value);
       });
-    },
-    setActive: (state) => {
-      // Toggle the pressed aria state component
-      Attribute.set(comp.element, 'aria-pressed', state);
-      // Toggle the inner button state, as that's the toggle component of the split button
-      SelectorFind.descendant(comp.element, 'span').each((button) => {
-        comp.getSystem().getByDom(button).each((buttonComp) => Toggling.set(buttonComp, state));
-      });
-    },
-    isActive: () => SelectorFind.descendant(comp.element, 'span').exists((button) => comp.getSystem().getByDom(button).exists(Toggling.isOn)),
-    setText: (text: string) =>
-      SelectorFind.descendant(comp.element, 'span').each((button) =>
-        comp.getSystem().getByDom(button).each((buttonComp) =>
-          AlloyTriggers.emitWith(buttonComp, updateMenuText, {
-            text
-          }))
-      ),
-    setIcon: (icon: string) =>
-      SelectorFind.descendant(comp.element, 'span').each((button) =>
-        comp.getSystem().getByDom(button).each((buttonComp) =>
-          AlloyTriggers.emitWith(buttonComp, updateMenuIcon, {
-            icon
-          }))
-      ),
+    }),
+    isActive: () => mainOpt.exists((c) => Toggling.isOn(c)),
+    setActive: (state: boolean) => mainOpt.each((c) => Toggling.set(c, state)),
     setTooltip: (tooltip: string) => {
-      const translatedTooltip = sharedBackstage.providers.translate(tooltip);
-      Attribute.set(comp.element, 'aria-label', translatedTooltip);
       tooltipString.set(tooltip);
+      mainOpt.each((c) => Attribute.set(c.element, 'aria-label', sharedBackstage.providers.translate(tooltip)));
+      chevronOpt.each((c) => Attribute.set(c.element, 'aria-label', sharedBackstage.providers.translate(`${tooltip} menu`)));
```

Comment:
Do we need full tooltips here? Or just the button's tooltip without the last used state like `Text color menu`?

Reply:
ah yes, it should be a static one.